sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobLength.java

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.specialized.cryptography;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.function.Function;
+
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V1;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V2;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_PROTOCOL_V2_1;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.NONCE_LENGTH;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.TAG_LENGTH;
+
+/**
+ * This class provides helper methods for adjusting encrypted downloads.
+ */
+final class EncryptedBlobLength {
+    private static final ClientLogger LOGGER = new ClientLogger(EncryptedBlobLength.class);
+
+    static Function<Long, Long> computeAdjustedBlobLength(EncryptionData encryptionData) {
```

Comment:
I think I missed this in the last review, but is there a particular reason we need to return a function here? Could we instead just return the long directly? I think it would be easier to follow the code if this was a long instead. It was sort of difficult following the `apply()` call that happens in the downstream ChunkedDownloadUtil logic.